smartsim/_core/generation/generator.py

```diff
@@ -249,10 +271,56 @@ def _build_tagged_files(tagged: TaggedFilesHierarchy) -> None:
 
             # write in changes to configurations
             if isinstance(entity, Model):
-                logger.debug(
-                    f"Configuring model {entity.name} with params {entity.params}"
+                files_to_params = self._writer.configure_tagged_model_files(
+                    to_write, entity.params
                 )
-                self._writer.configure_tagged_model_files(to_write, entity.params)
+                self._log_params(entity, files_to_params)
+
+    def _log_params(
+        self, entity: Model, files_to_params: t.Dict[str, t.Dict[str, str]]
+    ) -> None:
+        """Log which files were modified during generation
+
+        and what values were set to the parameters
+
+        :param entity: the model being generated
+        :type entity: Model
+        :param files_to_params: a dict connecting each file to its parameter settings
+        :type files_to_params: t.Dict[str, t.Dict[str, str]]
+        """
+        used_params: t.Dict[str, str] = {}
+        file_to_tables: t.Dict[str, str] = {}
+        for file, params in files_to_params.items():
+            used_params.update(params)
+            table = tabulate([[param, value] for param, value in params.items()],
```

Comment:
Seems like this is an unneeded cast to turn `dict_items` into a `list` when `tabulate` only requires an `Iterable`. This should be equivalent to simply
```py
tabulate(params.items(), headers=["Name", "Value"])
```
if you want to save the extra iteration

Reply:
I could be convinced that the extra cast to `list[list[str]]` makes this a hair more readable, and `dict_items` should be evaluated lazily, so its not like there is a performance hit here.

Feel free to leave "as is" if you see fit

Reply:
Done